### PR TITLE
M9: Build performance test report view

### PR DESCRIPTION
**Pull request by ninetteadhikari**
This PR does the following:

- adds d3 billboard wrapper to create chart
- restructure data to x and y-axis array. Converts test duration to minutes.
- adds zoom to the line charts
- updates measurement statistics data to include `start_time`
- updates the charts to use `start_time` as date for x-axis charts
- adds labels for x and y axis and tooltip
- adds section descriptions and styling updates
- updates default commit history length to 300

image

image

### CODE REVIEW

File: scripts/lib/build_perf/html/measurement_chart_d3billboard.html [hunk 1]
```diff
@@ -2,32 +2,77 @@
...
-    return Array.isArray(quantity) ? convertToMinute(quantity) : quantity
+  // Assuming the array values are durations in the format [hours, minutes, seconds, milliseconds]
+  const dataY = rawData.map(([commit, value]) => {
+    return Array.isArray(value) ? convertToMinute(value) : value
   });
-  dataY.unshift('quantity');
+  dataY.unshift('value');
+
+  let axisFormat
+  // Date is saved as string in the data
```
Comment (hulkoba) [inline scripts/lib/build_perf/html/measurement_chart_d3billboard.html:29]: Comments are important, but I think some of them can be removed. :)

File: scripts/lib/build_perf/html/measurement_chart_d3billboard.html [hunk 2]
```diff
@@ -2,32 +2,77 @@
...
+      y: {
+      label: {
+        text: "Disk size (MB)",
+        position: "outer-middle"
+      }
+    }
+    }
+  }
 
   // Example: https://naver.github.io/billboard.js/demo/#Chart.SimpleXYLineChart
```
Comment (hulkoba) [inline scripts/lib/build_perf/html/measurement_chart_d3billboard.html:66]: mhh

File: scripts/lib/build_perf/html/measurement_chart_d3billboard.html
Comment (hulkoba) [inline scripts/lib/build_perf/html/measurement_chart_d3billboard.html:1]: This is not a fix :) And can be part of the first commit

File: scripts/lib/build_perf/html/report.html
Comment (hulkoba) [inline scripts/lib/build_perf/html/report.html:1]: also not a fix :camel: 
Could you merge the style changes into the first commit and have this one only for descriptions?

File: scripts/lib/build_perf/html/measurement_chart_d3billboard.html
Comment (hulkoba) [inline scripts/lib/build_perf/html/measurement_chart_d3billboard.html:1]: What happened here? same commit twice?

File: scripts/oe-build-perf-report
Comment (hulkoba) [inline scripts/oe-build-perf-report:1]: commit history of what?

File: scripts/lib/build_perf/html/measurement_chart_d3billboard.html (hunk 2 above)
Comment (ninetteadhikari) [reply scripts/lib/build_perf/html/measurement_chart_d3billboard.html:66]: removed as I added apache echarts

File: scripts/lib/build_perf/html/measurement_chart_d3billboard.html
Comment (ninetteadhikari) [reply scripts/lib/build_perf/html/measurement_chart_d3billboard.html:1]: yea sorry about this 🙈 I need to update all the commit messages to remove 'fix' and combine some of them

File: scripts/oe-build-perf-report
Comment (ninetteadhikari) [reply scripts/oe-build-perf-report:1]: this is the commit history of the performance test shown in the charts. I increased the default from 25 to 300. This was one of the requirements in the Sow.

File: scripts/lib/build_perf/html/measurement_chart_d3billboard.html
Comment (ninetteadhikari) [reply scripts/lib/build_perf/html/measurement_chart_d3billboard.html:1]: hmm must have happened when I rebased it.. will clean up ☝️

**Pull request closed without merging**